cfpb/cfgov-refresh
# Ask Search bar

Adds Ask search bar

## Additions

- `o-search-bar` organism
- Search bar on each of the Ask pages
- `input-with-btn__bar` modifier
- `block__bg-branded` block modifier for green background
- `block__bg-extend-right` and `block__bg-extend-left` block modifiers to extend background offscreen
-`content__hide-horizontal-overflow` content modifier to hide background overflow
- Ask-specific searchbar css to customize bar display on different screen sizes for each page

## Testing

- Check out branch
- Set up Ask if necessary
  ```
  export DEPLOY_ENVIRONMENT=build
  ./drop-db.sh
  ./create-mysql-db.sh
  mysql -uroot v1 < ~/Downloads/production_django.sql
  python cfgov/manage.py migrate
  python cfgov/manage.py runscript migrate_knowledgebase
  ```
- Navigate to [ask landing page](http://localhost:8000/ask-cfpb/), [category page](http://localhost:8000/ask-cfpb/category-auto-loans/), [answer page](http://localhost:8000/ask-cfpb/what-effect-will-shopping-for-an-auto-loan-have-on-my-credit-en-763/), and [search results page](http://localhost:8000/ask-cfpb/search/?q=credit&btnSearch=) and verify that search bar appears and is functional

## Screenshots
__Landing page:__
![landing page](https://cloud.githubusercontent.com/assets/778171/26229785/03d0679c-3bf9-11e7-8b48-3c715da2145f.png)

__Answer page:__
![question page](https://cloud.githubusercontent.com/assets/778171/26229794/12921406-3bf9-11e7-8f24-3b914e34755b.png)

__Category page:__
![cat page](https://cloud.githubusercontent.com/assets/778171/26229798/1b3a99c0-3bf9-11e7-9743-803e500eaa54.png)

__Search results page:__
![search page](https://cloud.githubusercontent.com/assets/778171/26229803/219fb4ee-3bf9-11e7-9372-d3b86f1d8f57.png)





## Notes

- Search autocomplete will be added in separate PR
- `input-with-btn__bar` overrides most of the `input-with-btn` css so it could be its own class, but it's so similar in functionality that it seems useful to associate it as a modifier
- I added a flex box version of `input-with-btn__bar` for browsers that support it, but it's not really necessary



## Checklist
